projects/dex-ui/src/components/Liquidity/QuoteDetails.tsx

```diff
@@ -0,0 +1,155 @@
+import React, { useEffect, useMemo, useState } from "react";
+import styled from "styled-components";
+import { ERC20Token, Token, TokenValue } from "@beanstalk/sdk";
+import { formatEther } from "ethers/lib/utils";
+
+import gearIcon from "/src/assets/images/gear.svg";
+import infoIcon from "/src/assets/images/info.svg";
+import useSdk from "src/utils/sdk/useSdk";
+import { LIQUIDITY_OPERATION_TYPE, REMOVE_LIQUIDITY_MODE } from "./types";
+import { getPrice } from "src/utils/price/usePrice";
+
+type QuoteDetailsProps = {
+  type: LIQUIDITY_OPERATION_TYPE;
+  removeLiquidityMode?: REMOVE_LIQUIDITY_MODE | undefined;
+  quote:
+    | {
+        quote: TokenValue | TokenValue[];
+        estimate: TokenValue;
+      }
+    | null
+    | undefined;
+  slippage: number;
+  wellLpToken?: ERC20Token | undefined;
+  wellTokens?: Token[] | undefined;
+  slippageSettingsClickHandler: () => void;
+};
+
+const QuoteDetails = ({
+  type,
+  removeLiquidityMode,
+  quote,
+  slippage,
+  wellLpToken,
+  wellTokens,
+  slippageSettingsClickHandler
+}: QuoteDetailsProps) => {
+  const sdk = useSdk();
+  const [gasFeeUsd, setGasFeeUsd] = useState<string>("");
+
+  useEffect(() => {
+    const getGasInUsd = async () => {
+      const feeData = await sdk.provider.getFeeData();
+      const ethPrice = await getPrice(sdk.tokens.ETH, sdk)
+      const gBn = quote?.estimate.toBigNumber();
+      if (quote?.estimate && feeData.maxFeePerGas && gBn && ethPrice) {
+        const txEthAmount = gBn.mul(feeData.maxFeePerGas);
+        const txEthAmountNumber = formatEther(txEthAmount);
+        const usd = parseFloat(ethPrice.toHuman()) * parseFloat(txEthAmountNumber);
+        setGasFeeUsd(
+          `~${usd.toLocaleString("en-US", {
+            style: "currency",
+            currency: "USD"
+          })}`
+        );
+      }
+    };
```

Comment:
can we extract this to utils, we'll need gas estimation for swap too

Reply:
Also, line 47 and 48... can't you do all that through TokenValue? We can keep USD as a tokenvalue with 2 decimals. not 100% on this one, would need to try it out

Reply:
🤔 `TokenValue.mul` does not allow BigNumbers, which we get back from `feeData.maxFeePerGas`.